kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala

```diff
@@ -0,0 +1,236 @@
+  var role: KyuubiCtlActionRole = null
+  var zkAddress: String = null
+  var nameSpace: String = null
+  var user: String = null
+  var host: String = null
+  var port: String = null
+  var version: String = null
+  var verbose: Boolean = false
+
+  /** Default properties present in the currently defined default file. */
+  lazy val defaultKyuubiProperties: Map[String, String] = {
+    val maybeConfigFile = Utils.getDefaultPropertiesFile(env)
+    if (verbose) {
+      info(s"Using properties file: $maybeConfigFile")
+    }
+    val defaultProperties = Utils.getPropertiesFromFile(maybeConfigFile)
+    // Property files may contain sensitive information, so redact before printing
+    if (verbose) {
+      defaultProperties.foreach { case (k, v) =>
+        info(s"Adding default property: $k=$v")
+      }
+    }
+    defaultProperties
+  }
+
+  // Set parameters from command line arguments
+  parse(args.asJava)
+
+  // Use default property value if not set
+  useDefaultPropertyValueIfMissing()
+
+  validateArguments()
+
+  private def useDefaultPropertyValueIfMissing(): Unit = {
+    if (zkAddress == null) {
+      zkAddress = defaultKyuubiProperties.getOrElse(HA_ZK_QUORUM.key, null)
+    }
+    if (nameSpace == null) {
+      nameSpace = defaultKyuubiProperties.getOrElse(HA_ZK_NAMESPACE.key, null)
+    }
+    if (version == null) {
+      version = KYUUBI_VERSION
+    }
+  }
+  /** Ensure that required fields exists. Call this only once all defaults are loaded. */
+  private def validateArguments(): Unit = {
+    action match {
+      case KyuubiCtlAction.CREATE => validateActionArguments()
+      case KyuubiCtlAction.GET => validateActionArguments()
+      case KyuubiCtlAction.DELETE => validateActionArguments()
+      case KyuubiCtlAction.LIST => validateActionArguments()
+    }
+  }
+
+  private def validateActionArguments(): Unit = {
+    if (zkAddress == null) {
+      fail("Zookeeper address is not specified and no default value to load")
+    }
+    if (nameSpace == null) {
+      fail("Zookeeper namespace is not specified and no default value to load")
+    }
+    if (version == null) {
+      fail("Kyuubi version is not specified and could not found KYUUBI_VERSION in " +
+        "kyuubi-build-info")
+    }
+    if (host == null) {
+      fail("Must specify host for service")
+    }
+    if (port == null) {
+      fail("Must specify port for service")
+    }
+    if (role == KyuubiCtlActionRole.ENGINE && user == null) {
+      fail("Must specify user name for engine")
+    }
+  }
+
+  private def printUsageAndExit(exitCode: Int, unknownParam: Any = null): Unit = {
+    if (unknownParam != null) {
+      info("Unknown/unsupported param " + unknownParam)
+    }
+    val command = sys.env.getOrElse("_KYUUBI_CMD_USAGE",
+      """Usage: kyuubi-ctl <create|get|delete|list>  <server|engine> --zkAddress ...
+        |--namespace ... --user ... --host ... --port ... --version""".stripMargin)
+    info(command)
```

Comment:
Seems better to use `println`?

Reply:
I think it does not matter.

Normally, the info level is INFO,CONSOLE, user can see this in console.
Even the info level is ERROR, user can see below KyuubiCtlException message.

BTW, now I test it based on the logAppender.
If I found a solution to transfer printStream into it gracefully, I will use printStream to replace it.

Reply:
make sense